pkg/agent/watch.go

```diff
@@ -123,3 +123,30 @@ func makeVMEvent(vm *vmapi.VirtualMachine, kind vmEventKind) (vmEvent, error) {
 		podIP:   vm.Status.PodIP,
 	}, nil
 }
+
+// custom formatting for vmEvent so that it prints in the same way as VmInfo
+func (e vmEvent) Format(state fmt.State, verb rune) {
+	switch {
+	case verb == 'v' && state.Flag('#'):
+		state.Write([]byte(fmt.Sprintf(
+			// note: intentionally order podName and podIP before vmInfo because vmInfo is large.
+			"agent.vmEvent{kind:%q, podName:%q, podIP:%q, vmInfo:%#v}",
```

Comment:
this is prefixed as `api.VmInfo=` below... seems like we could be consistent there?

also just to confirm this is for controlling how things are printed to logs, or is there another consumer of this format?

Reply:
> also just to confirm this is for controlling how things are printed to logs, or is there another consumer of this format?

sorry, see the description of the PR now.